t/api/10-jobgroups.t

```diff
@@ -306,31 +423,25 @@ subtest 'prevent create/update duplicate job group on top level' => sub() {
 
 subtest 'prevent create parent/job group with empty or blank name' => sub() {
     for my $group (qw(parent_groups job_groups)) {
-        for my $name (undef, '   ') {
-            $t->post_ok("/api/v1/$group", form => {name => $name})->status_is(400);
-            is(
-                $t->tx->res->json->{error},
-                'The group name must not be empty or blank',
-                'Unable to create parent/job group with empty or blank name'
+        my %values = (missing => undef, invalid => '   ');
+        while (my ($error, $name) = each %values) {
```

Comment:
I wouldn't introduce a use of `each` here. Not sure what the Perl-gurus have to say about it but as far as I know it is slower than a regular for loop.

Reply:
It shouldn't be slower.
But for reproducible results I would iterate over the sorted keys instead ;-)
same below.

Reply:
2842I'll change it. Also #2842 for existing uses so we can make this consistent.